lib/iris/cube.py

```diff
@@ -1754,8 +1750,16 @@ def data(self):
 
     @data.setter
     def data(self, value):
-        if not (hasattr(value, 'shape') and hasattr(value, 'dtype')):
```

Comment:
This turns out to not be the complete check of what's required here. For example, a `numpy.int32` object has a `shape` attribute but is **not** an `ndarray`...